src/languages/kotlin.js

```diff
@@ -222,7 +229,7 @@ export default function(hljs) {
           {
             className: 'type',
             begin: /[,:]\s*/,
-            end: /[<\(,]|$/,
+            end: /[<\(,)]|$/,
```

Comment:
```suggestion
            end: /[<\(,)\s]|$/,
```

Since you're touching this to improve class detection, here's a minor finding. Not sure if this is the best way to fix it though.

Given the example code,

```kotlin
class MyView : View {
    constructor(ctx: Context) : super(ctx)
    constructor(ctx: Context, attrs: AttributeSet) : super(ctx, attrs)
}
```

I get this, (notice the `View {` being highlighted together)

![image](https://user-images.githubusercontent.com/1246453/174961886-2286f8a0-be22-4e03-8d23-bae6fb625347.png)

When it should be this,

![image](https://user-images.githubusercontent.com/1246453/174961823-c983db4c-56de-4d46-a9e7-2f416e9c0d43.png)